### DISCUSSION TITLE
Realflight doesnt recognize as joysticks

### DISCUSSION THREAD

**Discussion by rooster1973** [EdgeTX/edgetx#4009] [Q&A]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### What part of EdgeTX is the focus of this bug?

Transmitter firmware

### Current Behavior

Plug in and Realflight does not see the transmitter as a controller that is usable. tried other controller brands and types in setup with no change. previous release works fine.

### Expected Behavior

To detect transmitter as controller

### Steps To Reproduce

Plug in start up Realflight and select controller.

### Version

2.9.0

### Transmitter

Radiomaster TX16S / TX16SMK2

**Comment by raphaelcoeffic** [comment]
~~Is that `Realflight` under Windows or macOS?~~ I'm out, it's Windows only 😋

**Comment by rooster1973** [comment]
Sorry, under windows 11.

Justin

**Comment by raphaelcoeffic** [comment]
> Sorry, under windows 11

It seems RF needs some special adapter. Did it work with earlier EdgeTX versions?

**Comment by raphaelcoeffic** [comment]
This suggests it should work out of the box: https://steamcommunity.com/app/1070820/discussions/0/2966145784730214965/

```
As long as Windows recognizes your device as a USB game controller it will work in Realflight.
The need for a special dongle has been removed around version 8.
```

**Comment by 3djc** [comment]
Tested 2.9RC2 on tx16s, i get proper joystick movement under win11 josytick manager, I don't have Realflight to test with however

**Comment by rotorman** [comment]
@rooster1973 Press Win+R, enter `joy.cpl` and press Enter. A joystick dialog should open. Select the EdgeTX radio joystick and click on Properties. Do you see sliders move when you move the controls on the radio?

**Comment by pfeerick** [comment]
With Realflight Evolution, on Windows 11, "gamepad contollers" control panel detects my handset (running a 2.9-rc) correctly 
![image](https://github.com/EdgeTX/edgetx/assets/5500713/c62a61f6-8a47-4233-8254-17cebb8fc695)
and shows movement for the sticks and configured buttons. 
![image](https://github.com/EdgeTX/edgetx/assets/5500713/81089276-2c8f-4074-8694-68674ea8c4da)

Normally it will show a few more axes and buttons, but I'm using the new "advanced" mode to be able to configure buttons and switches to make Realflight work nicer with a radio handset. 

I did need to go into the Realflight settings after getting it to start up, and configure/map the controls... that was about it.

**Comment by 3djc** [comment]
My windows with default settings (deleted radio and models) show a ton more inputs 


![](https://media.discordapp.net/attachments/654988782403190784/1138763169804865629/image.png)

**Comment by rooster1973** [comment]
Yes it worked fine with last final release 2.8.4 I think.

Justin

**Comment by pfeerick** [comment]
1) Are you able to launch Realflight with the TX16S connected, and in joystick mode? (as you can't even start Realfight without joystick/controller connected). 
![image](https://github.com/EdgeTX/edgetx/assets/5500713/41b520ff-9be7-4c1e-8149-055a48318eec)

As this is otherwise working for me on TX16S with 2.9-RC2, with RF Evolution and Windows 11. :shrug:

2) With RealFlight running, after going into Settings -> Controller -> Configure Controller, what happens if you select "Blank Profile" and then "Calibrate"? Is movement detected? If it is, it is simply a matter of editing the profile, mapping the various functions to your preferred controller inputs, and then saving the profile... perhaps as "RM TX16S"?

**Comment by rooster1973** [comment]
I'll have to follow your directions and see if it works for me. I will have to update to rc 2 first.

Justin

**Comment by rooster1973** [comment]
Rc2 seems to work fine. Although im unable to setup switches for some reason. Doesn't recognize anything but main axis.

Justin

**Comment by pfeerick** [comment]
How many mixer lines do you have? You'll only have the controls made
available via the mixer (regardless of whether using classic or advanced
joystick mode).

**Comment by rooster1973** [comment]
I don't know I'm new to this radio and software and not sure what you're referring to.

Justin

**Comment by rotorman** [comment]
Open model settings and navigate to mixer tab.
If you are new to EdgeTX, it is advisable to work through following: https://edgetx.gitbook.io/edgetx-user-manual/
